File: JVM/jnetcall-java-common/src/main/java/jnetcall/java/client/model/CallState.java
package jnetcall.java.client.model;

import jnetbase.java.threads.ManualResetEvent;

public class CallState {

    public ManualResetEvent SyncWait;

    public ManualResetEvent AsyncWait;

    public Object Result;

    public void set() {
        if (SyncWait != null) SyncWait.set();
        if (AsyncWait != null) AsyncWait.set();
    }
}

File: NET/JNetCall.Sharp.Common/Client/Model/CallState.cs
﻿using System.Threading;
using Nito.AsyncEx;

namespace JNetCall.Sharp.Client.Model
{
    public class CallState
    {
        public ManualResetEvent SyncWait;

        public AsyncManualResetEvent AsyncWait;

        public object Result;

        public void Set()
        {
            SyncWait?.Set();
            AsyncWait?.Set();
        }
    }
}